sklearn/callback/tests/_utils.py

```diff
@@ -49,14 +50,15 @@ def __init__(self, max_iter=20, computation_intensity=0.001):
         self.max_iter = max_iter
         self.computation_intensity = computation_intensity
 
-    @_fit_context(prefer_skip_nested_validation=False)
-    def fit(self, X=None, y=None):
-        callback_ctx = self.__skl_init_callback_context__(
-            max_subtasks=self.max_iter
-        ).eval_on_fit_begin(estimator=self)
-
+    @_fit_callback
+    @_fit_context(prefer_skip_nested_validation=True)
+    def fit(self, X=None, y=None, X_val=None, y_val=None):
+        self._callback_fit_ctx.set_task_info(
```

Comment:
we can keep the variable to simplify the next references
```suggestion
        callback_ctx = self._callback_fit_ctx.set_task_info(
            task_name="fit", task_id=0, max_subtasks=self.max_iter
        )
```